client/signup/steps/page-picker/index.tsx

```diff
@@ -0,0 +1,233 @@
+	const selectedIndex = selectedPages.indexOf( pageId );
+	const totalSelections = selectedPages.length;
+	const isSelected = Boolean( selectedIndex > -1 );
+	const isDisabled = selectedIndex === -1 && totalSelections >= PAGE_LIMIT;
+	const title = useTranslatedPageTitles( pageId );
+
+	return (
+		<GridCellContainer
+			onClick={ () => onClick( pageId ) }
+			isSelected={ isSelected }
+			isClickDisabled={ isDisabled }
+		>
+			<BrowserView
+				pageId={ pageId }
+				isClickDisabled={ isDisabled }
+				isSelected={ isSelected }
+				selectedIndex={ selectedIndex >= 0 ? selectedIndex : -1 }
+			/>
+			<CellLabelContainer>
+				<div>{ title }</div>
+				{ popular ? <PopularContainer>{ translate( 'Popular' ) }</PopularContainer> : null }
+			</CellLabelContainer>
+		</GridCellContainer>
+	);
+}
+
+function PageSelector() {
+	const [ selectedPages, setSelectedPages ] = useState< string[] >( [
+		HOME_PAGE,
+		ABOUT_PAGE,
+		CONTACT_PAGE,
+	] );
+
+	const onPageClick = ( pageId: string ) => {
+		const foundIndex = selectedPages.indexOf( pageId );
+		// The home page cannot be touched
+		if ( pageId !== HOME_PAGE )
+			if ( foundIndex > -1 ) {
+				setSelectedPages( selectedPages.filter( ( page, index ) => index !== foundIndex ) );
+			} else if ( selectedPages.length !== PAGE_LIMIT ) {
+				setSelectedPages( [ ...selectedPages, pageId ] );
+			}
+	};
+
+	return (
+		<PageGrid>
+			<PageCell
+				popular
+				pageId={ HOME_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+			<PageCell
+				popular
+				pageId={ ABOUT_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+			<PageCell
+				popular
+				pageId={ CONTACT_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+
+			<PageCell
+				popular
+				pageId={ BLOG_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+
+			<PageCell
+				pageId={ PHOTO_GALLERY_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+			<PageCell
+				pageId={ SERVICE_SHOWCASE_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+			<PageCell
+				pageId={ VIDEO_GALLERY_PAGE }
+				selectedPages={ selectedPages }
+				onClick={ onPageClick }
+			/>
+			<PageCell pageId={ PODCAST_PAGE } selectedPages={ selectedPages } onClick={ onPageClick } />
+			<PageCell pageId={ PORTFOLIO_PAGE } selectedPages={ selectedPages } onClick={ onPageClick } />
+			<PageCell pageId={ FAQ_PAGE } selectedPages={ selectedPages } onClick={ onPageClick } />
+			<PageCell pageId={ SITEMAP_PAGE } selectedPages={ selectedPages } onClick={ onPageClick } />
+			<PageCell pageId={ PROFILE_PAGE } selectedPages={ selectedPages } onClick={ onPageClick } />
+		</PageGrid>
+	);
+}
+
+interface StepProps {
+	stepSectionName: string | null;
+	stepName: string;
+	goToStep: () => void;
+	goToNextStep: () => void;
+}
+
+const StyledButton = styled( Button )`
+	&.button.is-primary {
```

Comment:
If you remove the `compact` attribute from the button in L226, I think this extra styling may not be required.

Reply:
I kept in the padding to make it pixel perfect.